Sources/core/objects/Door.h

```diff
@@ -10,19 +10,19 @@ class Door: public IOnMapObject
     DECLARE_GET_TYPE_ITEM(Door);
     Door(size_t id);
     
+    enum STATE
+    {
+        OPEN, CLOSED, OPENING, CLOSING, WELDED  
+    };
+
     virtual void Open();
     virtual void Close();
     virtual void Process() override;
     virtual void Bump(IdPtr<IMovable> item) override;
     virtual void AttackBy(IdPtr<Item> item) override;
-    bool IsOpen() const { return door_state_ == OPEN; }
-    bool IsClosed() const { return door_state_ == CLOSED; }
+    bool CheckState(STATE state) { return state == door_state_ ; };
```

Comment:
It is better to name new function `IsState` because the function is simple and does not change internal state. `CheckSomething` assumes that something complicated can happen inside.
Also, it should be `const` function (like `IsOpen` or `IsClosed` before).